mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -630,6 +648,44 @@ impl Nexus {
         Ok(())
     }
 
+    /// get ANA state of the NVMe subsystem
+    pub async fn get_ana_state(&self) -> Result<NvmeAnaState, Error> {
+        if let Some(Protocol::Nvmf) = self.shared() {
+            if let Some(subsystem) = NvmfSubsystem::nqn_lookup(&self.name) {
+                let ana_state = subsystem.get_ana_state().await? as i32;
+                return match NvmeAnaState::from_i32(ana_state) {
+                    Some(state) => Ok(state),
+                    None => Err(Error::InvalidNvmeAnaState {
+                        ana_value: ana_state,
+                    }),
```

Comment:
I guess it's a matter of personal preference, but you can use Option::ok_or_else() here to convert an Option to a Result

Reply:
Thanks, `rust-lint` came back with "error: unnecessary closure used to substitute value for `Option::None`" and suggested `ok_or` instead so I've used that.